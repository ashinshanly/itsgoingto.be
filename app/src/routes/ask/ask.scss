@import "variables";

.header-container {
  padding-top: 6.25em;
  margin-bottom: 3.125em;
}

.header {
  margin-bottom: 2em;

  h1 {
    font-size: 4em;
    line-height: 1.2em;
    min-height: 1.2em;
    letter-spacing: -1px;
    font-weight: 300;
    color: $base-font-color;

    margin-bottom: 0px;

    span {
      font-weight: 500;
    }
  }
}
