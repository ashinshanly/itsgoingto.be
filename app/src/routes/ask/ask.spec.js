/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Ask } from './ask'

const props = {
  question: '',
  hasQuestion: false,
  canSubmitPoll: false,
  poll: {},
  answers: [],
  clearPoll: jest.fn(),
}

describe('(Route) Ask', () => {
  describe('(Action) componentDidMount', () => {
    it('should call prop clearPoll', () => {
      const wrapper = shallow(<Ask {...props} />)

      wrapper.instance().componentDidMount()
      expect(props.clearPoll).toHaveBeenCalled()
    })
  })
  describe('(Render)', () => {
    describe('with question', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Ask {...props} question={'Some question'} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when hasQuestion is true', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Ask {...props} hasQuestion={true} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when canSubmitPoll is true', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Ask {...props} canSubmitPoll={true} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('with poll', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Ask {...props} poll={{ question: 'Question', identifier: 'kdH98eJ' }} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('with answers', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Ask {...props} answers={['Answer 1', 'Answer 2']} />)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
