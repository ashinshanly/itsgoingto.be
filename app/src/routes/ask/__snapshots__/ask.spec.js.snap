// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) Ask (Render) when canSubmitPoll is true matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      It's Going To Be
    </title>
    <meta
      content="Wondering where to go or what to see? Start a poll and share it with your friends or colleagues."
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div
    className="container header-container hideable"
  >
    <div
      className="header center-text"
    >
      <h1>
        <WordRotate
          words="What,Where,When,Who"
        />
         is it going to be?
      </h1>
    </div>
  </div>
  <div
    className="container question-container"
  >
    <Connect(Question)
      canSubmitPoll={true}
      placeholderText={
        Array [
          "What film should we watch?",
          "Who is going to win the league?",
          "Where should we go for drinks?",
          "You talking to me?",
          "What should we do this weekend?",
          "When should we go to Paris?",
          "Who ya gonna call?",
          "When will you start a poll?",
        ]
      }
      question=""
    />
    <Component
      answers={Array []}
      hasQuestion={false}
    />
    <Options
      hasQuestion={false}
      poll={Object {}}
    />
    <Connect(Actions)
      canSubmitPoll={true}
      hasQuestion={false}
    />
  </div>
</div>
`;

exports[`(Route) Ask (Render) when hasQuestion is true matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      It's Going To Be
    </title>
    <meta
      content="Wondering where to go or what to see? Start a poll and share it with your friends or colleagues."
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div
    className="container header-container hideable gone"
  >
    <div
      className="header center-text"
    >
      <h1>
        <WordRotate
          words="What,Where,When,Who"
        />
         is it going to be?
      </h1>
    </div>
  </div>
  <div
    className="container question-container move-up"
  >
    <Connect(Question)
      canSubmitPoll={false}
      placeholderText={
        Array [
          "What film should we watch?",
          "Who is going to win the league?",
          "Where should we go for drinks?",
          "You talking to me?",
          "What should we do this weekend?",
          "When should we go to Paris?",
          "Who ya gonna call?",
          "When will you start a poll?",
        ]
      }
      question=""
    />
    <Component
      answers={Array []}
      hasQuestion={true}
    />
    <Options
      hasQuestion={true}
      poll={Object {}}
    />
    <Connect(Actions)
      canSubmitPoll={false}
      hasQuestion={true}
    />
  </div>
</div>
`;

exports[`(Route) Ask (Render) with answers matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      It's Going To Be
    </title>
    <meta
      content="Wondering where to go or what to see? Start a poll and share it with your friends or colleagues."
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div
    className="container header-container hideable"
  >
    <div
      className="header center-text"
    >
      <h1>
        <WordRotate
          words="What,Where,When,Who"
        />
         is it going to be?
      </h1>
    </div>
  </div>
  <div
    className="container question-container"
  >
    <Connect(Question)
      canSubmitPoll={false}
      placeholderText={
        Array [
          "What film should we watch?",
          "Who is going to win the league?",
          "Where should we go for drinks?",
          "You talking to me?",
          "What should we do this weekend?",
          "When should we go to Paris?",
          "Who ya gonna call?",
          "When will you start a poll?",
        ]
      }
      question=""
    />
    <Component
      answers={
        Array [
          "Answer 1",
          "Answer 2",
        ]
      }
      hasQuestion={false}
    />
    <Options
      hasQuestion={false}
      poll={Object {}}
    />
    <Connect(Actions)
      canSubmitPoll={false}
      hasQuestion={false}
    />
  </div>
</div>
`;

exports[`(Route) Ask (Render) with poll matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      It's Going To Be
    </title>
    <meta
      content="Wondering where to go or what to see? Start a poll and share it with your friends or colleagues."
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div
    className="container header-container hideable"
  >
    <div
      className="header center-text"
    >
      <h1>
        <WordRotate
          words="What,Where,When,Who"
        />
         is it going to be?
      </h1>
    </div>
  </div>
  <div
    className="container question-container"
  >
    <Connect(Question)
      canSubmitPoll={false}
      placeholderText={
        Array [
          "What film should we watch?",
          "Who is going to win the league?",
          "Where should we go for drinks?",
          "You talking to me?",
          "What should we do this weekend?",
          "When should we go to Paris?",
          "Who ya gonna call?",
          "When will you start a poll?",
        ]
      }
      question=""
    />
    <Component
      answers={Array []}
      hasQuestion={false}
    />
    <Options
      hasQuestion={false}
      poll={
        Object {
          "identifier": "kdH98eJ",
          "question": "Question",
        }
      }
    />
    <Connect(Actions)
      canSubmitPoll={false}
      hasQuestion={false}
    />
  </div>
</div>
`;

exports[`(Route) Ask (Render) with question matches snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      It's Going To Be
    </title>
    <meta
      content="Wondering where to go or what to see? Start a poll and share it with your friends or colleagues."
      name="description"
    />
    <meta
      content="question vote poll result"
      name="keywords"
    />
  </HelmetWrapper>
  <div
    className="container header-container hideable"
  >
    <div
      className="header center-text"
    >
      <h1>
        <WordRotate
          words="What,Where,When,Who"
        />
         is it going to be?
      </h1>
    </div>
  </div>
  <div
    className="container question-container"
  >
    <Connect(Question)
      canSubmitPoll={false}
      placeholderText={
        Array [
          "What film should we watch?",
          "Who is going to win the league?",
          "Where should we go for drinks?",
          "You talking to me?",
          "What should we do this weekend?",
          "When should we go to Paris?",
          "Who ya gonna call?",
          "When will you start a poll?",
        ]
      }
      question="Some question"
    />
    <Component
      answers={Array []}
      hasQuestion={false}
    />
    <Options
      hasQuestion={false}
      poll={Object {}}
    />
    <Connect(Actions)
      canSubmitPoll={false}
      hasQuestion={false}
    />
  </div>
</div>
`;
